text/0000-iri-v181-transaction-and-bundle/0000-iri-v181-transaction-and-bundle.md

```diff
@@ -0,0 +1,980 @@
+            head_transaction.tag = transaction_draft.obsolete_tag;
+        }
+        head_transaction.set_nonce_from_proof_of_work(proof_of_work);
+
+        // `next_local_trunk` is the hash that will be used as value of the current transaction pointing to the previous
+        // one.
+        let mut next_local_trunk = head_transaction.calculate_hash(sponge);
+
+        // Set the rest of the transactions
+        for transaction_draft in builder_iterator {
+            transaction_draft.branch(trunk_tip);
+            transaction_draft.trunk(branch_tip);
+
+            transaction_draft.attachment_timestamp = timestamp();
+            transaction_draft.attachment_timestamp_lower = 0;
+            transaction_draft.attachment_timestamp_upper = 3812798742493;
+
+            if transaction_draft.tag.is_empty() {
+                transaction_draft.tag = transaction_draft.obsolete_tag;
+            }
+
+            transaction_draft.set_nonce_from_proof_of_work(proof_of_work);
+            next_local_trunk = transaction_draft.calculate_hash(sponge);
+        }
+    }
+
+    /// Signs those transaction drafts contained in the builder that withdraw funds from an address.
+    ///
+    /// Only the owner of the seed from which the source address is generated has the right to remove funds from it. The
+    /// seed is thus used to generate the signature, which in turn is inserteed in those transaction drafts that withdraw
+    /// funds. Bundles that contain withdrawing transactions that have missing or bad signatures will be considered invalid
+    /// and thus rejected.
+    ///
+    /// TODO: This function relies on some `SignatureScheme` trait, which is not yet defined. The code below lays out
+    ///       how the signing scheme is thought to be used.
+    ///
+    /// TODO: This function relies on some `Wallet` trait, which is not yet defined. It's supposed to provide some
+    ///       `f: Wallet W -> Tx Address A -> Signature Scheme Inputs I`
+    pub fn sign<I, S, W>(self, signature_scheme: S, seed: S::Seed, wallet: Wallet) -> Result<Self, OutgoingBundleError>
+    where
+        S: SignatureScheme<I>,
+        W: Wallet<A, Output=S::I>,
+    {
+        unimplemented!();
+
+        // Pseudocode how `sign` should be used.
+
+        let mut current_index = 0
+
+        // TODO: Explain what this loop is trying to achieve.
+        for transaction_draft in self {
+            if transaction_draft.value < 0 {
+                if transaction_draft.current_index >= current_index {
+                    let input = inputs[transaction_draft.address];
+
+                    // NOTE: The specific arguments to the signature scheme depend on the implementation of `SignatureScheme`.
+                    //
+                    // NOTE: The specific arguments to wallet depend on the implementation of `Wallet`.
+                    let fragments = transaction_draft.sign(signature_scheme, seed, wallet.get(transaction_draft.address));
```

Comment:
It's not on the `transaction_draft` scope. Will be on private key scope from Signing Scheme.

Reply:
I have made a note of this in the sample code.